File: c/LeetCode_50_Pow.c
#include <stdio.h>


double myPow(double x, int n) {
    int is_negative = (n < 0) ? 1 : 0;
    n = abs(n);
    int res = 1;
    while (n > 0) {
        if (n % 2 == 1) {
            res *= x;
        }
        x *= x;
        n /= 2;
    }

    return (is_negative) ? 1 / res : res;
}


int main() {
    myPow(2, -2);
    return 0;
}

File: python/LeetCode_50_Pow.py
def myPow(x: float, n: int) -> float:
    is_negative = 1 if n < 0 else 0
    n = abs(n)

    res = 1
    while (n > 0):
        if n % 2 == 1:
            res *= x
        x *= x
        n //= 2

    return 1 / res if is_negative else res 
    

myPow(2., 10)